### PR TITLE
Add context to error messages

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Adding more context to error messages printed to the build log. 

### Changes

- Add prefix to error messages in calls to `failf()`
- Regenerate README

### CODE REVIEW

**Review by shams-ahmed — DISMISSED** [review]

File: main.go [hunk 1]
```diff
@@ -159,38 +159,38 @@ func main() {
 
 		_, err := version.NewVersion(config.NDKVersion)
 		if err != nil {
-			failf(fmt.Sprintf("'%s' is not a valid NDK version. This should be the full version number, such as 23.0.7599858. To see all available versions, run 'sdkmanager --list'", config.NDKVersion))
+			failf(fmt.Sprintf("install dependencies: '%s' is not a valid NDK version. This should be the full version number, such as 23.0.7599858. To see all available versions, run 'sdkmanager --list'", config.NDKVersion))
```
Comment (godrei) [inline main.go:162]: `install dependencies` is for the Step's dependency installation phase.
Here, these installations are part of the `run` phase.

Comment (ofalvai) [reply main.go:162]: Are you sure? What is the intended use of the `install dependencies` prefix? Things like installing Ruby gems for eg. Cocoapods?

Comment (godrei) [reply main.go:162]: Yes, here is an example: https://github.com/bitrise-steplib/steps-xcode-test/blob/master/main.go#L38

Comment (ofalvai) [reply main.go:162]: Fixed, take a look now @godrei

**Review by tothszabi — APPROVED** [review]

**Pull request merged**